main.tf

```diff
@@ -32,7 +36,7 @@ resource "null_resource" "helm_install" {
       users:
       - name: aws
         user:
-          token: ${var.token}
+          token: ${data.aws_eks_cluster_auth.cluster.token}
```

Comment:
Why can't we get the token from the output of eks module. Its not super important but the terraform module should keep only the things specific to itself.

Also is this done because the token expires ?

Reply:
we should also derive `${var.cluster_endpoint}` and `${var.cluster_ca_certificate}` from the same data source

Reply:
> Why can't we get the token from the output of eks module. Its not super important but the terraform module should keep only the things specific to itself.

I checked before, token is not passed in outputs of eks module, neither from us, and nor from aws module

Reply:
we can use this
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#attribute-reference
 for fetching `cluster_endpoint` and `ca_certificate`, 

 doing that 
 once in main.tf -> passing it in helm
 
Or 

do data block call in helm everytime we use helm 

P.s We're also making data block part of modules.yaml,  similar to modules section, we will have a data block section in yaml, 
since, we also require aws data block to fetch account id from current profile. ( making the end user do less work)